src/scribe_data/cli/cli_utils.py

```diff
@@ -52,6 +62,129 @@
+    """
+    extraction_dir = LANGUAGE_DATA_EXTRACTION_DIR
+    available_languages = []
+    for lang_folder in extraction_dir.iterdir():
+        if lang_folder.is_dir():  # Check if it's a directory
+            lang_name = lang_folder.name
+            sparql_file_path = lang_folder / "verbs" / "query_verbs.sparql"
+            qid = extract_qid_from_sparql(sparql_file_path)
+            if qid:
+                available_languages.append((lang_name, qid))
+            else:
+                available_languages.append((lang_name, ""))
+    return available_languages
+
+
+def extract_qid_from_sparql(file_path: Path) -> str:
+    """
+    Extract the QID from the specified SPARQL file.
+
+    Args:
+        file_path (Path): Path to the SPARQL file.
+
+    Returns:
+        str | None: The extracted QID or None if not found.
+    """
+    try:
+        with open(file_path, "r", encoding="utf-8") as file:
+            content = file.read()
+            # Use regex to find the QID (e.g., wd:Q34311)
+            match = re.search(r"wd:Q\d+", content)
+            if match:
+                return match.group(0).replace("wd:", "")  # Return the found QID
+    except Exception as _:
+        pass
+        # print(f"Error reading {file_path}: {e}")
+    return None  # Return None if not found or an error occurs
+
+
+def check_and_update_languages():
+    """
+    Check for missing languages in the metadata and update if necessary.
+    """
+    available_languages = get_available_languages()
+    existing_languages = {
+        lang["language"].lower() for lang in language_metadata["languages"]
+    }
+    missing_languages = [
+        lang
+        for lang in available_languages
+        if lang[0].lower() not in existing_languages
+    ]
+    if missing_languages:
+        update_language_metadata(missing_languages)
+
+
+def update_language_metadata(missing_languages: list[tuple[str, str]]):
+    """
+    Update the language metadata with missing languages.
+
+    Args:
+        missing_languages (list[tuple[str, str]]): Missing languages and their QIDs.
+
+    Returns:
+        None
+    """
+    try:
+        with LANGUAGE_METADATA_FILE.open("r+", encoding="utf-8") as file:
+            language_metadata = json.load(file)
+
+            for lang in missing_languages:
+                language_metadata["languages"].append(
+                    {"language": lang[0].lower(), "qid": lang[1]}
+                )
+            # Move the file pointer to the beginning and overwrite the existing file
+            file.seek(0)
+            json.dump(language_metadata, file, ensure_ascii=False, indent=4)
+            file.truncate()  # Remove any leftover data
+        print("Language metadata updated successfully.")
+    except (IOError, json.JSONDecodeError) as e:
+        print(f"Error updating language metadata: {e}")
+
+
+def set_metadata(language_name: str, qid: str):
```

Comment:
If we do keep `set_metadata`, then we should include an iso argument :)

Reply:
Also @DeleMike, it'd be great if for those functions that are kept that you check some of the other docstrings and use similar formatting. As the function docstrings are written now, they won't be parsed correctly for the readthedocs documentation :)

Reply:
oh! I did not know about this. will update :)